### PR TITLE
chore: apply rate limit to ws

### PR DESCRIPTION
**Pull request by RetricSu**
```sh
   {
        "id": 1,
        "jsonrpc": "2.0",
        "error": {
            "code": -32005,
            "message": "Too Many Requests, IP: ::ffff:127.0.0.1, please wait 23s and retry. RPC method: web3_clientVersion."
        }
    }
```

### CODE REVIEW

File: packages/api-server/src/rate-limit.ts [hunk 1]
```diff
@@ -88,6 +101,35 @@ export async function rateLimit(
...
+    const message = `Too Many Requests, IP: ${reqId}, please wait ${remainSecs}s and retry. RPC method: ${rpcMethod}.`;
+    const error: JSONRPCError = {
+      code: LIMIT_EXCEEDED,
+      message: message,
+    };
+
+    logger.debug(
+      `Rate Limit Exceed, ip: ${reqId}, method: ${rpcMethod}, ttl: ${remainSecs}s`
+    );
+    return { error, remainSecs };
```
Comment (keroro520) [inline packages/api-server/src/rate-limit.ts:126]: What about throwing the error?

**Review by keroro520 — APPROVED** [review]

File: packages/api-server/src/rate-limit.ts (hunk 1 above)
Comment (RetricSu) [reply packages/api-server/src/rate-limit.ts:126]: throwing requires some refactor on http/ws, maybe leave it to future PRs

File: packages/api-server/src/rate-limit.ts [hunk 2]
```diff
@@ -88,6 +101,35 @@ export async function rateLimit(
   return isBan;
 }
 
+export async function _rateLimit(
```
Comment (classicalliu) [inline packages/api-server/src/rate-limit.ts:104]: Rename this method without `_` prefix?

Comment (RetricSu) [reply packages/api-server/src/rate-limit.ts:104]: we have a method `rateLimit` already, this two can be refactor in the future

Comment (classicalliu) [reply packages/api-server/src/rate-limit.ts:104]: I means don't start with `_`, in our eslint rule, `_` prefix means ignore

Comment (RetricSu) [reply packages/api-server/src/rate-limit.ts]: updated

**Review by classicalliu — APPROVED** [review]

**Pull request merged**